### ISSUE TITLE
Output JSON format issue

### ISSUE THREAD

**Issue by zhlmmc** [promptfoo/promptfoo#130]
Based on my test, I found when there is only one provider, the output JSON will not show the provider name in results:
![image](https://github.com/promptfoo/promptfoo/assets/1726527/fdb0f73c-3854-42b9-93f2-97a0c08daae3)

I'm not sure why it is designed like this. I think it is more reliable to show the provider all the times.
![image](https://github.com/promptfoo/promptfoo/assets/1726527/ba16f5c3-dd13-40f3-a7d8-e8da7ff94fd2)

**Comment by typpo** [comment]
If an eval is run on a single provider, I think the information is redundant atop every single column.  Is there a reason you want it displayed?

**Comment by zhlmmc** [comment]
> If an eval is run on a single provider, I think the information is redundant atop every single column. Is there a reason you want it displayed?

When parse the output, I have the following code:
```TypeScript
/**
 * Find all results of provided llm and extract the scores
 */
function getLLMScores(llm_id: string, results) {
    var scores: TestScore[] = [];
    for (const result of results) {
        // currently, when there is only one provider, promptfoo's return result will not display provider id
        // TODO
        if (result.prompt.display.includes(llm_id)) {
            var score: TestScore = {
                test_name: result.vars.name,
                score: result.score
            }
            scores.push(score);
        }
    }
    return scores;
}
```
As there is no field in results showing which provider generated this result, I can only get the information from field "display" which is a workaround. Do you think it is necessary to add provider field in each of the result?

**Comment by typpo** [comment]
I see.  Change b6104e9a65c417e4b83a0ea74919d3c77dfa24eb will include the provider id in `result.provider.id`, which should be a cleaner way to accomplish what you're trying to do!

**Comment by typpo** [comment]
This change is published in 0.23.0.
